datastore-design/go-grpc-demo/storagebackend/timescaledb/getobservations.go

```diff
@@ -4,37 +4,39 @@ import (
 	"database/sql"
 	"datastore/datastore"
 	"fmt"
-
 	_ "github.com/lib/pq"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"math"
 )
 
 // retrieveObsForTS retrieves into obs observations for time series tsID in
 // open-ended range [fromTime, toTime>.
 // Returns nil upon success, otherwise error.
 func retrieveObsForTS(
-	db *sql.DB, tsID int64, fromTime, toTime int64, obs *[]*datastore.Observation) error {
+	db *sql.DB, tsID int64, fromTime *timestamppb.Timestamp, toTime *timestamppb.Timestamp, obs *[]*datastore.Observation) error {
```

Comment:
Suggest simplifying to `... fromTime, toTime *timestamppb.Timestamp ...`

Reply:
Apologies, that was my lack of understanding of Go. Changed.